pkg/nvcdi/api.go

```diff
@@ -44,4 +44,33 @@ const (
 	// HookEnableCudaCompat refers to the hook used to enable CUDA Forward Compatibility.
 	// This was added with v1.17.5 of the NVIDIA Container Toolkit.
 	HookEnableCudaCompat = HookName("enable-cuda-compat")
+	// HookCreateSymlinks refers to the hook used create symlinks inside the
+	// directory path to be mounted into a container.
+	HookCreateSymlinks = HookName("create-symlinks")
+	// HookUpdateLDCache refers to the hook used to  Update the dynamic linker
+	// cache inside the directory path to be mounted into a container.
+	HookUpdateLDCache = HookName("update-ldcache")
 )
+
+// NewHookName takes a string and returns a []HookName, empty if the HookName
```

Comment:
This is not the right place to implement this. we have a `pkg/nvcdi/hooks.go` file where this can go instead.

Also, in terms of the implementation, does adding special handling to the `disabledHooks` type for the `all` hook not make this cleaner?